sql/dd/impl/bootstrap/bootstrapper.cc

```diff
@@ -1352,7 +1353,9 @@ bool initialize_dd_properties(THD *thd) {
   */
   if (!bootstrap::DD_bootstrap_ctx::instance().is_initialize() &&
       bootstrap::DD_bootstrap_ctx::instance().is_dd_upgrade() &&
-      update_system_tables(thd)) {
+      update_system_tables(thd) &&
+      update_properties(thd, nullptr, nullptr,
```

Comment:
By using this update_properties() the DD_properties of the dictionary tables are stored on disk? Next restart shouldn't depend on in-memory flag?

Also why do we need this unconditionally?

Reply:
Am guessing that restart would keep using this in-memory flag and add double "encryption=y" again? Have to debug and find out

Reply:
DD properties after update_properties is like this:

p definition->raw_string()
$8 = "fields=elem0\\=def\\\\\\=metadata BLOB NOT NULL\\\\\\;lbl\\\\\\=metadata\\\\\\;pos\\\\\\=2\\\\\\;\\;elem1\\=def\\\\\\=table_id BIGINT UNSIGNED NOT NULL\\\\\\;lbl\\\\\\=table_id\\\\\\;pos\\\\\\=0\\\\
\\;\\;elem2\\=def\\\\\\=version BIGINT UNSIGNED NOT NULL\\\\\\;lbl\\\\\\=version\\\\\\;pos\\\\\\=1\\\\\\;\\;;foreign_keys=;indexes=elem0\\=def\\\\\\=PRIMARY KEY (table_id)\\\\\\;lbl\\\\\\=index_pk\\\\\\;p
os\\\\\\=0\\\\\\;\\;;name=innodb_dynamic_metadata;options=elem0\\=def\\\\\\=DEFAULT CHARSET\\\\\\\\\\\\\\=utf8\\\\\\;lbl\\\\\\=CHARSET\\\\\\;pos\\\\\\=1\\\\\\;\\;elem1\\=def\\\\\\=COLLATE\\\\\\\\\\\\\\=ut
f8_bin\\\\\\;lbl\\\\\\=COLLATION\\\\\\;pos\\\\\\=2\\\\\\;\\;elem2\\=def\\\\\\=ENCRYPTION\\\\\\\\\\\\\\='Y'\\\\\\;lbl\\\\\\=ENCRYPTION\\\\\\;pos\\\\\\=6\\\\\\;\\;elem3\\=def\\\\\\=ENGINE\\\\\\\\\\\\\\=INNO
DB\\\\\\;lbl\\\\\\=ENGINE\\\\\\;pos\\\\\\=0\\\\\\;\\;elem4\\=def\\\\\\=ROW_FORMAT\\\\\\\\\\\\\\=DYNAMIC\\\\\\;lbl\\\\\\=ROW_FORMAT\\\\\\;pos\\\\\\=3\\\\\\;\\;elem5\\=def\\\\\\=STATS_PERSISTENT\\\\\\\\\\\\
\\=0\\\\\\;lbl\\\\\\=STATS_PERSISTENT\\\\\\;pos\\\\\\=4\\\\\\;\\;elem6\\=def\\\\\\=TABLESPACE\\\\\\\\\\\\\\=mysql\\\\\\;lbl\\\\\\=TABLESPACE\\\\\\;pos\\\\\\=5\\\\\\;\\;;"

We can see that ENCRYPTION="Y". So this goes to disk. Next I have to check if next restart also adds the same